### PR TITLE
Refactoring IRemoteStorage to fix encapsulation issues

### PR DESCRIPTION
**Pull request by Jharaxus**
Fixes the encapsulation issues of IRemoteStorage and improve its interface.

### CODE REVIEW

File: app/src/main/java/com/github/polypoly/app/menu/JoinGameLobbyActivity.kt [hunk 1]
```diff
@@ -567,6 +566,6 @@ class JoinGameLobbyActivity : ComponentActivity() {
 private val authenticated_user = User(7, "current_user", "", Skin(0, 0, 0),
     Stats(0, 0, 0, 0, 0), listOf(), mutableListOf()
 )
-private val mockDb = FakeRemoteStorage()
+private val mockDb = com.github.polypoly.app.network.FakeRemoteStorage()
```
Comment (AmLibra) [inline app/src/main/java/com/github/polypoly/app/menu/JoinGameLobbyActivity.kt:569]: Could probably be left out (tested and compiles)

File: app/src/main/java/com/github/polypoly/app/network/FakeRemoteStorage.kt [hunk 2]
```diff
@@ -74,56 +77,68 @@ class FakeRemoteStorage : IRemoteStorage {
...
         return CompletableFuture.completedFuture(true)
     }
 
-    override fun updateGameLobby(gameLobby: GameLobby): Future<Boolean> {
-        mockGameLobbies[gameLobby.code] = gameLobby
+    override fun <T> setValue(key: String, value: T): CompletableFuture<Boolean> {
+        if (value == null)
+            throw java.lang.IllegalArgumentException("null value")
+        datas[key] = value
         return CompletableFuture.completedFuture(true)
```
Comment (AmLibra) [inline app/src/main/java/com/github/polypoly/app/network/FakeRemoteStorage.kt:142]: Agreed upon and will be refactored for next PR

File: app/src/main/java/com/github/polypoly/app/network/IRemoteStorage.kt [hunk 3]
```diff
@@ -1,75 +1,79 @@
...
-     */
-    fun updateGameLobby(gameLobby: GameLobby): Future<Boolean>
-}
+/**
+ * Extension function of IRemoteStorage::getAllValues to enable prettier call of the function
+ */
+inline fun <reified T : Any> IRemoteStorage.getValue(key: String) = getValue(key, T::class)
 
 enum class StorageType {
     FIREBASE,
```
Comment (AmLibra) [inline app/src/main/java/com/github/polypoly/app/network/IRemoteStorage.kt:79]: Good usage of Generic types

**Review by AmLibra — APPROVED** [review]
Overall high quality code. Some issues regarding test code coverage and possible refactoring, but everything should be fixed by the next PR.

File: app/src/main/java/com/github/polypoly/app/menu/JoinGameLobbyActivity.kt (hunk 1 above)
Comment (Jharaxus) [reply app/src/main/java/com/github/polypoly/app/menu/JoinGameLobbyActivity.kt:569]: Fixed !

File: app/src/main/java/com/github/polypoly/app/menu/ProfileModifyingActivity.kt [hunk 4]
```diff
@@ -37,7 +39,7 @@ class ProfileModifyingActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         val id = intent.getLongExtra("userId", 0)
-        val user = FakeRemoteStorage.instance.getUserWithId(id).get()
+        val user = FakeRemoteStorage.instance.getValue<User>(DB_USERS_PROFILES_PATH + id).get()
```
Comment (AlSchlo) [inline app/src/main/java/com/github/polypoly/app/menu/ProfileModifyingActivity.kt:42]: Calling `.get()` on every future kind of beats the whole purpose of having futures...

Comment (AlSchlo) [reply app/src/main/java/com/github/polypoly/app/menu/ProfileModifyingActivity.kt:42]: This applies everywhere of course

**Pull request merged**